tests_e2e/tests/hibernation/hibernation.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import uuid
+
+from assertpy import fail
+from typing import List, Dict, Any
+
+from azurelinuxagent.common import conf
+from azurelinuxagent.common.utils.timeutil import create_utc_timestamp
+
+from tests_e2e.tests.lib.agent_log import AgentLog
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.agent_test_context import AgentVmTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachine
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+
+
+class Hibernation(AgentVmTest):
+    """
+    A hibernate/resume cycle is a special case in that resume produces a new Fabric goal state with incarnation 1. Since the VM is re-allocated,
+    that goal state will include a new tenant encryption certificate. If the incarnation was also 1 before hibernation, the Agent won't detect
+    this new goal state and subsequent Fast Track goal states would fail because they require the new certificate.
+
+    The Agent has logic to detect this scenario and fetch the required certificates; this test verifies that logic.
+
+    See https://learn.microsoft.com/en-us/azure/virtual-machines/linux/hibernate-resume-linux for details on hibernation
+    """
+    def __init__(self, context: AgentVmTestContext):
+        super().__init__(context)
+        self._ssh_client: SshClient = self._context.create_ssh_client()
+
+    def run(self):
+        log.info("Executing test on %s - IP address: %s", self._context.vm, self._context.ip_address)
```

Comment:
could we add a new line after this to make the test logs more readable?
`log.info("")`